compiler/rustc_mir_build/src/thir/cx/expr.rs

```diff
@@ -703,8 +703,7 @@ impl<'tcx> Cx<'tcx> {
                                 tcx,
                                 anon_const.def_id.to_def_id(),
                             )
-                            .instantiate_identity()
-                            .normalize(tcx, self.param_env);
+                            .instantiate_identity();
```

Comment:
I think this is just using `identity_unevaluated` because that's a helper that exists on `Const` to create a `Const::Unevaluated`... but it has a binder we don't need so it gets stripped again... or is there a good reason for this `instantiate_identity` business?

Reply:
It's best practice to return `EarlyBinder`s for people to understand that they need to instantiate their possibly unsubstituted consts. I wouldn't touch this in this PR.

Reply:
This is about the binders of the DefId?

Reply:
The binder corresponds to the unsubstituted generic params of the generic. The fact we `instantiate_identity` is just acknowledging that we're calling this const constructor on a const that comes from the same set of generics that we're building this MIR body in.

Reply:
> Would it make sense to rename this to from_unevaluated? I don't understand the point of the "identity" in the function name.

Yeah, I don't really care about the name. The identity is a bit redundant given the `EarlyBinder`. 👍